### PR TITLE
copier: init at 7.0.1

### PR DESCRIPTION
**Pull request by Artturin**
Rebased https://github.com/NixOS/nixpkgs/pull/145528 wrong

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/poetry-dynamic-versioning/default.nix [hunk 1]
```diff
@@ -5,6 +5,7 @@
 , jinja2
 , markupsafe
 , poetry-core
+, poetry
```
Comment (dotlambda) [inline pkgs/development/python-modules/poetry-dynamic-versioning/default.nix:8]: This will be removed from `pythonPackages`: #209534

Comment (Artturin) [reply pkgs/development/python-modules/poetry-dynamic-versioning/default.nix:8]: the tests want to run poetry commands

**Comment by dotlambda** [comment]
Why was this merged into staging-next?

**Comment by Artturin** [comment]
> Why was this merged into staging-next?

Staging-next had the required python updates last I checked

**Comment by dotlambda** [comment]
> > Why was this merged into staging-next?
> 
> Staging-next had the required python updates last I checked

But it was merged into master before you merged the PR.

File: pkgs/development/python-modules/pyyaml-include/default.nix [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-9/vrjnG1C+Dm4HRy98edv7GhW63pyToHg2n/SeV+Z3E=";
+  };
+
+  nativeBuildInputs = [
+    pyyaml
+    setuptools-scm
+    setuptools-scm-git-archive
+    toml
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pyyaml-include/default.nix:22]: I assume this is for setuptools-scm[toml] which is always included in setuptools-scm.

**Comment by SuperSandro2000** [comment]
This should have gone to master but mistakes happen. Not a big problem IMO.

**Pull request merged**